cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/DatasetSystemMetadataWriter.java

```diff
@@ -114,6 +116,9 @@ public DatasetSystemMetadataWriter(MetadataStore metadataStore,
       dataset instanceof InputFormatProvider || dataset instanceof OutputFormatProvider) {
       tags.add(BATCH_TAG);
     }
+    if (dsProperties.getProperties().containsKey(Constants.AppFabric.WORKFLOW_LOCAL_DATASET_PROPERTY)) {
```

Comment:
If the value of `dsProperties.getProperties().get(Constants.AppFabric.WORKFLOW_LOCAL_DATASET_PROPERTY)` is a boolean, then it would be a good idea to check if the value is `true` before adding the tag.